explainer.md

```diff
@@ -345,6 +341,114 @@ And here's the result:
 
 https://user-images.githubusercontent.com/93594/184120371-678f58b3-d1f9-465b-978f-ee5eab73d120.mp4
 
+## Cross-document same-origin transitions
+
+This section outlines the navigation specific aspects of the ViewTransition API. The rendering model for generating snapshots and dispalying them using a tree of targetable pseudo-elements is the same for both SPA/MPA.
+
+### Declarative opt-in to transitions
+
+The first step is to add a new meta tag to the old and new Documents. This tag indicates that the author wants to enable transitions for same-origin navigations to/from this Document.
+
+```html
+<meta name="view-transition" content="same-origin">
+```
+
+The above is equivalent to the browser implicitly executing the following script in the SPA API:
+
+```js
+document.startViewTransition(() => updateDOMToNewPage());
+```
+
+This results in a cross-fade between the 2 Documents from the default CSS set up by the browser. The transition executes only if this tag is present on both the old and new Documents. The tag must also be added before the body element is parsed.
+
+The motivation for a declarative opt-in, instead of a script event, is:
+
+* Enabling authors to define transitions with no script. If the transition doesn't need to be customized based on the old/new URL, it can be defined completely in CSS.
+
+* Avoiding undue latency in the critical path for browser initiated navigations like back/forward. We want to avoid dispatch of a script event for each of these navigations.
+
+Issue: The meta tag can be used to opt-in to other navigation types going forward: same-document, same-site, etc.
+
+### Script Events
+
+Script can be used to customize a transition based on the URL of the old/new Document; or the current state of the Document when the transition is initiated. The Document could've been updated since first old from user interaction.
+
+#### Script on Old Document
+```js
+document.addEventListener("viewTransitionOnOldDocument", (event) => {
+  // Cancel the transition (based on new URL) if needed.
+  if (shouldNotTransition(event.url)) {
```

Comment:
Is `event.url` after redirects? I wonder if something in the navigation API is a good place for this.

Reply:
Yes. The implementation I have waits to have all redirects resolve and get a response from the server for the new Document before dispatching this event (and initiating snapshoting). There is a trade-off here:

- Snapshot too early and the current Document's rendering freezes/it becomes inert too soon. We can't keep rendering frames for it for the same rendering suppression problem that SPA has.
- Snapshot too late and we won't parallelize all the snapshoting work with the network fetch of the new page.

We'll get a better idea of optimizing this with some demos etc.

Reply:
Given the discussion below for a feature specific event here, can we resolve this?